acados/ocp_nlp/ocp_nlp_dynamics_cont.c

```diff
@@ -499,6 +526,30 @@ void ocp_nlp_dynamics_cont_update_qp_matrices(void *config_, void *dims_, void *
         blasfeo_dveccp(nx1, mem->pi, 0, &mem->adj, nu + nx);
     }
 
+    if (opts->compute_hess)
+    {
+        int hess_index = 0;
+        for (int j = 0; j < nx; j++) {
+            for (int k = j; k < nx; k++) {
+                BLASFEO_DMATEL(&mem->hes, nu+k, nu+j) = work->sim_out->S_hess[hess_index];
+                hess_index++;
+            }
+            for (int k = 0; k < nu; k++) {
+                // S: nu x nx
+                BLASFEO_DMATEL(&mem->hes, nu+j, k) = work->sim_out->S_hess[hess_index];
+                hess_index++;
+            }
+        }
+        for (int j = 0; j < nu; j++) {
+            for (int k = j; k < nu; k++) {
+                BLASFEO_DMATEL(&mem->hes, k, j) = work->sim_out->S_hess[hess_index];
+                hess_index++;
+            }
+        }
+
+        // Add hessian contribution
+        blasfeo_dgead(nx+nu, nx+nu, 1.0, &mem->hes, 0, 0, mem->RSQrq, 0, 0);
+    }
```

Comment:
Nice! But I guess you are using the Hessian provided by the ERK integrator, right? According to @FreyJo this Hessian is not correct, did you two interact maybe on this topic?

Reply:
@roversch yes as @dkouzoup  wrote. Maybe you can check my `sim_test_hessian` if I am using them wrong, or compare with the one provided by IRK after it is merged.

Reply:
@FreyJo, did you in the end do any changes in the ERK integrator for the Hessians? Can you compare them again Robin's changes and see what makes sense here?

Reply:
yes I did, see my comment below.
IMO: `expl_vde_adj` is sometimes used for hessian generation, even though it seems to be intended to be used if only adjoints are being propagated. With my fix, this "misuse" is still possible, i.e. nothing broke.
With my fix `expl_ode_hes` and  `expl_vde_adj` are used equally (which I guess is not as originally intended but needed to not break some examples). The difference is that `expl_ode_hes` when `opts->sens_hess == true` and otherwise `expl_vde_adj` is used.

Reply:
We tried it quickly, but there seems to be a conflict here. The Hessians of @FreyJo breaks my exact Hessian example. We will investigate later.